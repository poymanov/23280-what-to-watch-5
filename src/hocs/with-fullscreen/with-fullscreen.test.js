import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withFullscreen from "./with-fullscreen";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withFullscreen(MockComponent);

it(`withFullscreen is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
