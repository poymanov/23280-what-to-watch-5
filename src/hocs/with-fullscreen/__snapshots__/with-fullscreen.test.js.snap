// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withFullscreen is rendered correctly 1`] = `
<div
  className="FullScreen"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div />
</div>
`;
